rosidl_typesupport_introspection_cpp/resource/msg_TypeSupport_Introspection.cpp.template

```diff
@@ -19,6 +19,26 @@
 #include "rosidl_generator_cpp/MessageTypeSupport.h"
 #include "rosidl_typesupport_introspection_cpp/FieldTypes.h"
 #include "rosidl_typesupport_introspection_cpp/MessageIntrospection.h"
+#include "rosidl_typesupport_introspection_cpp/impl/visibility_control.h"
+
+@{
+import os
+
+IS_WINDOWS = os.name == 'nt'
+
+if not IS_WINDOWS:
+    print('#ifndef offsetof')
+    print('  #define offsetof __builtin_offsetof')
+    print('#endif')
```

Comment:
It would be good to explicitly include it, but that clearly isn't required since `offsetof` is defined already on OS X. It's probably implicitly pulled in by one of the other headers already.